presto-manager-agent/src/main/java/com/teradata/prestomanager/agent/LogFilter.java

```diff
@@ -0,0 +1,288 @@
+     *
+     * Builder instances can be safely reused.
+     */
+    public static class Builder
+    {
+        private Path file;
+        private Pattern logPattern;
+        private String lineSeparator = System.getProperty("line.separator");
+        private ImmutableMap.Builder<String, Predicate<String>> namedGroupFilters = ImmutableMap.builder();
+        private ImmutableMap.Builder<Integer, Predicate<String>> numberedGroupFilters = ImmutableMap.builder();
+        private Integer maxEntries;
+        private boolean keepFirst = false;
+
+        private Builder() {}
+
+        public LogFilter build()
+                throws FileNotFoundException
+        {
+            return new LogFilter(file, logPattern, lineSeparator,
+                    namedGroupFilters.build(),
+                    numberedGroupFilters.build(),
+                    maxEntries, keepFirst);
+        }
+
+        public Builder setFile(Path file)
+        {
+            requireNonNull(file);
+            this.file = file;
+            return this;
+        }
+
+        /**
+         * Use the given regular expression to match each line of a log file
+         */
+        public Builder setPattern(Pattern pattern)
+        {
+            requireNonNull(pattern);
+            this.logPattern = pattern;
+            return this;
+        }
+
+        /**
+         * Add a filter to apply to a named capturing group
+         * in the regular expression for log file lines
+         */
+        public Builder addGroupFilter(String groupName, Predicate<String> filter)
+        {
+            requireNonNull(groupName);
+            if (filter != null) {
+                namedGroupFilters.put(groupName, filter);
+            }
+            return this;
+        }
+
+        /**
+         * Add a filter to apply to a capturing group in
+         * the regular expression for log file lines
+         */
+        public Builder addGroupFilter(Integer groupNumber, Predicate<String> filter)
+        {
+            requireNonNull(groupNumber);
+            if (filter != null) {
+                numberedGroupFilters.put(groupNumber, filter);
+            }
+            return this;
+        }
+
+        /**
+         * Use this string to separate lines in the same entry
+         */
+        public Builder setLineSeparator(String separator)
+        {
+            requireNonNull(separator);
+            lineSeparator = separator;
+            return this;
+        }
+
+        /**
+         * Set the maximum number of entries to keep when filtering.
+         *
+         * A null capacity results in all entries being kept.
+         */
+        public Builder setCapacity(Integer capacity)
+        {
+            maxEntries = capacity;
+            return this;
+        }
+
+        /**
+         * When there is a maximum capacity, keep the first entries
+         * instead of the last entries
+         */
+        public Builder keepFirst()
```

Comment:
Maybe consider adding `unkeepFirst` method as well (or simply `setKeepFirst(boolean keepFirst)`) in case you want to allow `Builder` changing the behavior, e.g.
```
Builder builder;
builder.keepFirst();
if (...) {
builder.unKeepFirst();
}
```